src/mbgl/annotation/annotation_tile.hpp

```diff
@@ -45,7 +45,10 @@ class AnnotationTileLayer : public GeometryTileLayer {
     AnnotationTileLayer(std::string);
 
     std::size_t featureCount() const override { return features.size(); }
-    std::unique_ptr<GeometryTileFeature> getFeature(std::size_t i) const override { return std::make_unique<AnnotationTileFeature>(features[i]); }
+    std::unique_ptr<GeometryTileFeature> getFeature(std::size_t i) const override {
+        assert(i < features.size());
+        return std::make_unique<AnnotationTileFeature>(features[i]);
```

Comment:
Instead of adding an assertion, let's change `features[i]` to `features.at(i)`. The latter has bounds-checking built in, and is what `VectorTileLayer::getFeature` [uses](https://github.com/mapbox/mapbox-gl-native/blob/bce352bd86f7faf98b3ba5bbb31d3a7860533199/src/mbgl/tile/vector_tile.cpp#L313).